substrate/frame/contracts/src/xcm.rs

```diff
@@ -0,0 +1,138 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{AccountIdOf, Config, Error};
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{pallet_prelude::DispatchResultWithPostInfo, weights::Weight};
+use frame_system::pallet_prelude::BlockNumberFor;
+use sp_runtime::{DispatchError, DispatchResult};
+use xcm::{VersionedMultiLocation, VersionedXcm};
+use xcm_executor::traits::QueryResponseStatus;
+
+pub mod adapter;
+
+pub use adapter::XCMAdapter;
+pub type CallOf<T> = <T as frame_system::Config>::RuntimeCall;
+
+pub trait XCM<T: Config> {
+	type QueryId: Encode + Decode + MaxEncodedLen;
+	type WeightInfo: WeightInfo;
+
+	/// Execute an XCM message locally. see [`pallet_xcm::pallet::Pallet::execute`]
```

Comment:
```suggestion
	/// Execute an XCM locally. see [`pallet_xcm::pallet::Pallet::execute`]
```

Remember, what does the M in XCM stand for?

Reply:
ha yes good point 😓